vmaas/reposcan/database/csaf_store.py

```diff
@@ -1,178 +1,378 @@
+
+            file_id = self.cve2file_id[cve]
+            to_upsert = [
+                (row[0], id_, status_id, file_id)
+                for id_, status_id in products.to_tuples(("id_", "status_id"), with_id=True)
+            ]
+            execute_values(
+                cur,
+                """
+                    INSERT INTO csaf_cve_product (cve_id, csaf_product_id, csaf_product_status_id, csaf_file_id)
                     VALUES %s
-                    ON CONFLICT (cve, csaf_product_id)
-                    DO UPDATE SET product_status_id = EXCLUDED.product_status_id;
-                    """, [(cve, product[0], product[1]) for product in products])
-                self.conn.commit()
+                    ON CONFLICT (cve_id, csaf_product_id)
+                    DO UPDATE SET csaf_product_status_id = EXCLUDED.csaf_product_status_id
+                """,
+                to_upsert,
+            )
+            self.conn.commit()
         except Exception as exc:
             CSAF_FAILED_IMPORT.inc()
             CSAF_FAILED_UPDATE.inc()
-            self.logger.exception("Failure while inserting or updating data to csaf_cves: %s", exc)
+            self.logger.exception("%s: ", FAILED_CVE_UPSERT)
             self.conn.rollback()
+            raise CsafStoreException(FAILED_CVE_UPSERT) from exc
         finally:
             cur.close()
 
-    def _remove_cves(self, cve, ids):
-        if not ids:
+    def _remove_cves(self, cve: str, products: model.CsafProducts) -> None:
+        if not products:
             return
+
+        cur = self.conn.cursor()
         try:
-            cur = self.conn.cursor()
-            cur.execute("""DELETE FROM csaf_cves WHERE cve = %s and csaf_product_id not in %s""",
-                        (cve, tuple(row[1] for row in ids)))
+            cur.execute(
+                """
+                    DELETE FROM csaf_cve_product ccp
+                    USING cve
+                    WHERE ccp.cve_id = cve.id
+                        AND cve.name = %s
+                        AND csaf_product_id NOT IN %s
+                """,
+                (cve, tuple(x.id_ for x in products)),
+            )
             self.conn.commit()
         except Exception as exc:
             CSAF_FAILED_DELETE.inc()
-            self.logger.exception("Failure while removing data from csaf cves: %s", exc)
+            self.logger.exception("%s: ", FAILED_CVE_DELETE)
             self.conn.rollback()
+            raise CsafStoreException(FAILED_CVE_DELETE) from exc
         finally:
             cur.close()
 
-    def _map_products_by_status(self, products) -> dict[int, list[tuple[str, str, str]]]:
-        products_by_status = defaultdict(list)
+    def _populate_cves(self, csaf_cves: model.CsafCves) -> None:
+        for cve, products in csaf_cves.items():
+            products_copy = deepcopy(products)  # only for logging of failed cves
+            try:
+                self._update_product_ids(products)
```

Comment:
Trying with empty database and after this function is called, `products` is always empty list which results that no products are ever inserted on the next line.

Reply:
did you have any cpes and packages in db before trying the sync? if cpe for the product is missing in db then the product is skipped, cpes are not inserted as part of the csaf sync. Same for the packages, if the package is missing then the product (and CVE) is skipped

Reply:
I've synced CPEs but seems they are not matched... Looking

Reply:
Discussing privately, issue seems more complicated